stellargraph/layer/graph_attention.py

```diff
@@ -843,6 +843,10 @@ def __init__(
                 )
             )
 
+        self.node_model = deprecated_model_function(self._node_model, "node_model")
```

Comment:
Sorry to re-review before being suggested, but my original suggestion was careful to create these statically in each class. To be more specific, this would be like

```python
class GAT:
    ...
    def _node_model(self):
        ...

    node_model = deprecated_model_function(_node_model, "node_model")
    link_model = deprecated_model_function(_link_model, "link_model")
    build = deprecated_model_function(in_out_tensors, "build")
```

This approach should mean they're still shown in the documentation, unlike dynamically assigning them when building each instance. After doing this, and moving the doc string from `deprecated_model_function` to `_function_wrapper`, the docs look like:


![image](https://user-images.githubusercontent.com/1203825/77481936-de202180-6e78-11ea-81ab-bcf43ce0eb3e.png)

Compared to the current approach of this PR of https://stellargraph--1140.org.readthedocs.build/en/1140/api.html#stellargraph.layer.graph_attention.GAT.in_out_tensors which just shows `in_out_tensors`.

Reply:
nice good point!

Reply:
@huonw my current approach isn't triggering deprecation warnings, can you see anything wrong?

Reply:
> my current approach isn't triggering deprecation warnings, can you see anything wrong?

What do you mean by this specifically? How are you testing?

Per #1117, the `stacklevel` needs to be set correctly (although it looks like it is), and, even with this, they'll only show up in Jupyter, or in script run with Python 3.7 not Python 3.6.

Reply:
I'm testing some models in `test_deprecated_model_function` in `test_misc.py`, and I've also tried using the depracted functions in jupyter notebooks. The functions work but I'm not getting a warning

Reply:
i'm getting jupyter warnings now nevermind, must just be my test